open_schools_platform/organization_management/circles/services.py

```diff
@@ -105,3 +113,71 @@ def is_organization_related_to_student_profile(organization_id: str, student_pro
 def convert_str_to_point(string: str):
     res = re.findall(r"\d+\.\d+", string)
     return Point(float(res[0]), float(res[1]), srid=4326)
+
+
+def setup_scheduled_notifications(circle: Circle, notification_delays: list[timedelta]):
+    tasks = PeriodicTask.objects.filter(args=f'["{circle.id}"]', task=send_circle_lesson_notification.name)
+    CrontabSchedule.objects.filter(id__in=list(map(lambda task: task.crontab.id, tasks))).delete()
+
+    if circle.start_time is None:
+        tasks.delete()
+        return
+
+    cron_list = list(map(lambda delta: create_crontab_schedule(circle, delta), notification_delays))
+    if len(tasks) == 0 or len(tasks) != len(notification_delays):
+        PeriodicTask.objects.filter(id__in=list(map(lambda task: task.id, tasks))).delete()
+        create_periodic_tasks(circle, cron_list)
+    else:
+        for i in range(len(cron_list)):
+            tasks[i].crontab = cron_list[i]
+            tasks[i].name = f'teacher notification [{cron_list[i]}_{circle.id}]'
```

Comment:
we should make special constant for task name
I would add app name in this name, is it reasonable?